Lib/pdb.py

```diff
@@ -1315,7 +1315,12 @@ def do_commands(self, arg):
         reached.
         """
         if not arg:
-            bnum = len(bdb.Breakpoint.bpbynumber) - 1
+            for bnum in range(len(bdb.Breakpoint.bpbynumber) - 1, -1, -1):
+                if bdb.Breakpoint.bpbynumber[bnum] is not None:
+                    break
+                if bnum == 0:
+                    self.error('no breakpoints set')
+                    return
```

Comment:
We can using the `for ... else ...` syntax to make the code more readable.

```suggestion
            else:
                self.error('no breakpoints set')
                return
```

Reply:
See also https://github.com/python/cpython/pull/142835#discussion_r2623943070.